deb/build/debian/jenkins.init

```diff
@@ -43,6 +45,26 @@ if [ "$RUN_STANDALONE" = "false" ]; then
     exit 1
 fi
 
+# Make sure there exists a java executable, it may not be allways the case
+if [ -z "$JAVA" ]; then
+    echo "ERROR: No Java executable found in current PATH: $PATH" >&2
+    echo "If you actually have java installed on the system make sure the executable is in the aforementioned path and that 'which java' returns the java executable path" >&2
+    exit 1
+fi
+
+# Work out the JAVA version we are working with:
+JAVA_VERSION=$($JAVA -version 2>&1 | sed -n ';s/.* version "\(.*\)\.\(.*\)\..*"/\1\2/p;')
+
+if [ "$JAVA_VERSION" = "$JAVA_ALLOWED_VERSION" ]; then
```

Comment:
Well, Jenkins project does not define which JDKs it supports. We do not test with IBM Java, but one may run with it if he wants. Some known issues: https://issues.jenkins-ci.org/issues/?jql=labels%20%3D%20java-ibm